react/features/dtmf/components/DialDtmfDialog.js

```diff
@@ -0,0 +1,203 @@
+// @flow
+
+import { FieldTextStateless } from '@atlaskit/field-text';
+import React, { Component } from 'react';
+
+import { sendTones } from '../../base/conference';
+import { Dialog } from '../../base/dialog';
+import { translate } from '../../base/i18n';
+import { connect } from '../../base/redux';
+
+declare var APP: Object;
+
+type Props = {
+
+    /**
+     * Invoked to obtain translated strings.
+     */
+    t: Function
+};
+
+/**
+ * The type of the React {@code Component} state of {@link DialDtmfDialog}.
+ */
+type State = {
+
+    /**
+     * The currently entered number sequence.
+     */
+    numbers: string,
+
+    /**
+     * The currently entered tone duration (ms).
+     */
+    toneDuration: number,
+
+    /**
+     * The currently entered pause duration (ms).
+     */
+    pauseDuration: number
+};
+
+/**
+ * A dialog that allows to dial a DTMF number sequence.
+ */
+class DialDtmfDialog extends Component<Props, State> {
```

Comment:
This doesn't seem like too complex of a component, would it be possible to refactor it into a function component? We're trying to move towards a more functional style, especially in simpler use cases.